test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc

```diff
@@ -531,5 +532,13 @@ TEST_F(FileUtilsTest, ReadFileContents) {
   EXPECT_EQ(data, output);
 }
 
+TEST_F(QuicPlatformTest, TestEnvoyQuicBufferAllocator) {
+  QuicStreamBufferAllocator allocator;
+  char* p = allocator.New(1024);
+  EXPECT_NE(nullptr, p);
+  memset(p, 'a', 1024);
+  allocator.Delete(p);
```

Comment:
You could also in this test ensure that the total allocated bytes was the same before and after this delete, via `Memory::Stats::totalCurrentlyAllocated()` before and after. See test/common/stats/thread_local_store_test.cc , test HeapStatsThreadLocalStoreTest.MemoryWithoutTls for an example. Only do that test if TestUtil::hasDeterministicMallocStats() is true.